libraries/extensions/ros2-bridge/msg-gen/src/parser/package.rs

```diff
@@ -102,15 +104,63 @@ fn get_ros_msgs_each_package<P: AsRef<Path>>(root_dirs: &[P]) -> Result<Vec<Pack
             deps.insert("action_msgs".to_owned());
         }
         deps.retain(|dep| map.contains_key(dep));
-        package.dependencies = deps
+        dependencies.insert(package.name.clone(), deps);
+    }
+
+    // flatten the dependencies
+    let dependencies = {
+        let mut flattened_dependencies = HashMap::<String, BTreeSet<String>>::new();
+        let mut finished_set: BTreeSet<String> = dependencies.keys().cloned().collect();
+        while let Some(first) = finished_set.first().cloned() {
+            dfs_collect(
+                &first,
+                &dependencies,
+                &mut flattened_dependencies,
+                &mut finished_set,
+            );
+        }
+        flattened_dependencies
+    };
+
+    for package in &mut packages {
+        package.dependencies = dependencies
+            .get(&package.name)
+            .unwrap()
             .iter()
-            .filter_map(|dep| map.get(dep).and_then(|pkg| Some(pkg.clone())))
+            .filter_map(|dep| map.get(dep).map(|pkg| pkg.clone()))
             .collect();
     }
 
     Ok(packages)
 }
 
+fn dfs_collect(
```

Comment:
Could be cool to have a quick description or better function name as I have truly no idea what dfs refers to from a quick glance.